activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -317,12 +317,12 @@ def async_pick(*column_names)
       async.pick(*column_names)
     end
 
-    # Pluck all the ID's for the relation using the table's primary key
+    # Pluck all the base model's ID's for the relation using the table's primary key
     #
     #   Person.ids # SELECT people.id FROM people
     #   Person.joins(:companies).ids # SELECT people.id FROM people INNER JOIN companies ON companies.person_id = people.id
     def ids
-      pluck primary_key
+      records.pluck(primary_key)
```

Comment:
`records` call `load`, so this impact performance in two ways:

  - It will perform a `SELECT *`. so fetch more data.
  - It will instantiate a model for each row.

So I don't think this solution is acceptable.

Reply:
Thanks for the review @casperisfine.
I'll see if I can find a better solution.